hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/config/EvmConfiguration.java

```diff
@@ -147,43 +179,134 @@ CacheManager cacheManagerRecordFileEarliest() {
+
+        MainnetPrecompiledContracts.populateForIstanbul(precompileContractRegistry, gasCalculator);
+
+        final Map<String, Provider<MessageCallProcessor>> messageCallProcessors = Map.of(
+                EVM_VERSION_0_30, () -> messageCallProcessor, EVM_VERSION_0_34, () -> mirrorEvmMessageCallProcessor);
+
         return new MirrorEvmTxProcessorImpl(
                 worldState,
                 pricesAndFees,
                 evmProperties,
                 gasCalculator,
-                mcps(
-                        gasCalculator,
-                        autoCreationLogic,
-                        entityAddressSequencer,
-                        mirrorEvmContractAliases,
-                        evmProperties,
-                        precompileMapper,
-                        basicHbarCentExchange,
-                        prngSystemPrecompiledContract,
-                        prngSeedOperation),
-                ccps(gasCalculator, evmProperties, prngSeedOperation),
+                messageCallProcessors,
+                contractCreationProcessors,
                 blockMetaSource,
                 mirrorEvmContractAliases,
                 abstractCodeCache,
                 mirrorOperationTracer,
                 store);
     }
+
+    @Bean
+    CreateOperationExternalizer createOperationExternalizer() {
+        return new CreateOperationExternalizer() {
+            @Override
+            public void externalize(final MessageFrame frame, final MessageFrame childFrame) {
+                // do nothing
+            }
+
+            @Override
+            public boolean shouldFailBasedOnLazyCreation(final MessageFrame frame, final Address contractAddress) {
+                return false;
+            }
+        };
+    }
+
+    @Bean
+    EVM evm(
+            final GasCalculatorHederaV22 gasCalculator,
+            final MirrorNodeEvmProperties mirrorNodeEvmProperties,
+            final HederaPrngSeedOperation prngSeedOperation,
+            final CreateOperationExternalizer createOperationExternalizer) {
+        final var operationRegistry = new OperationRegistry();
+        final BiPredicate<Address, MessageFrame> validator = (Address x, MessageFrame y) -> true;
+
+        registerShanghaiOperations(
+                operationRegistry,
+                gasCalculator,
+                mirrorNodeEvmProperties.chainIdBytes32().toBigInteger());
+        Set.of(
+                        new HederaBalanceOperation(gasCalculator, validator),
+                        new HederaDelegateCallOperation(gasCalculator, validator),
+                        new HederaEvmChainIdOperation(gasCalculator, mirrorNodeEvmProperties),
+                        new HederaEvmCreate2Operation(
+                                gasCalculator, mirrorNodeEvmProperties, createOperationExternalizer),
+                        new HederaEvmCreateOperation(gasCalculator, createOperationExternalizer),
```

Comment:
nit: Can avoid the extra param. Spring is smart enough to only create one.

```suggestion
                                gasCalculator, mirrorNodeEvmProperties, createOperationExternalizer()),
                        new HederaEvmCreateOperation(gasCalculator, createOperationExternalizer()),
```